smac/tae/execute_func.py

```diff
@@ -172,11 +177,37 @@ def run(self, config, instance=None,
             runtime = time.time() - start_time
             additional_run_info = {}
 
+        # check serializability of results - if not, then attempt to convert it
+        cost = self.serialize(cost)
+        additional_run_info = self.serialize(additional_run_info)
+
         return status, cost, runtime, additional_run_info
 
     def _call_ta(self, obj, config, instance, seed):
         raise NotImplementedError()
 
+    def _call_ta_raw(self, config, **kwargs):
```

Comment:
I think this function needs to go to the two subclasses to allow both the dict and the list-like version to work.

Reply:
This function is not needed actually. The `_call_ta` method can be reused here. I'll update it.